src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/Whitelist.java

```diff
@@ -66,6 +68,79 @@
 
     public abstract boolean permitsStaticFieldSet(@Nonnull Field field, @CheckForNull Object value);
 
+    // Utility methods for creating canonical string representations of the signature
+    public static final StringBuilder joinWithSpaces(StringBuilder b, String[] types) {
```

Comment:
`joinWithSpaces` and `getName` (both variants) and `argumentTypes` are all just relocated from EnumeratingWhitelist unchanged (allowing them to be reused here).

Reply:
Note:  I'm retaining these here with a shim in `EnumeratingWhitelist` because these much more logically belong here -- providing a canonical identifier.  Also useful if we do another round of optimization or add another implementation, because these Strings make logical caching keys.